File: app_project/frontend/src/models/policy.ts
export type PolicyType = "p" | "g";

export interface PolicyItem {
  ptype: PolicyType;   // "p" = policy, "g" = grouping
  v0: string;          // subject: user/role
  v1: string;          // object: resource/menu
  v2?: string;         // action: view/create/etc
  v3?: string;
  v4?: string;
  v5?: string;
}

export interface PolicyCreate {
  ptype: PolicyType;
  v0: string;
  v1: string;
  v2?: string;
}

File: app_project/backend/app/schemas/settings/policy.py
from typing import Optional
from pydantic import BaseModel

class PolicyItem(BaseModel):
    ptype: str
    v0: str
    v1: str
    v2: Optional[str] = None
    v3: Optional[str] = None
    v4: Optional[str] = None
    v5: Optional[str] = None

class PolicyCreate(BaseModel):
    ptype: str
    v0: str
    v1: str
    v2: Optional[str] = None
